subsys/spm/secure_services.c

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#include <cortex_m/tz.h>
+#include <misc/reboot.h>
+
+#if defined(CONFIG_SPM_SECURE_SERVICES)
+
+/** @brief Implementation of Secure System Reset
+ *
+ * Currently, this simply calls the Zephyr System Reset function,
+ * issuing a system reboot immediately.
+ *
+ * Note: the function will be located in a Non-Secure
+ * Callable region of the Secure Firmware Image.
+ */
+static void spm_system_reboot(void)
+{
+	sys_reboot(SYS_REBOOT_WARM);
```

Comment:
But, is this function always compiled in, @oyvindronningstad ?
Maybe I miss something, but it looks to me that it reboot.c is requires CONFIG_REBOOT

Reply:
Yes, it's enabled in the Kconfig now.

Reply:
Wouldn't it be nice to guard the function with CONFIG_REBOOT?

Reply:
Like this?

    #ifdef CONFIG_REBOOT
    static void spm_system_reboot(void)
    {
    	sys_reboot(SYS_REBOOT_WARM);
    }
    #endif

Reply:
Well, yeap something like that.
Do you think it makes sense? We might later add more services into this file , but we might want to build the SPM with a subset of those, and the _reboot might not be in that set....

Reply:
I imagine we might add separate Kconfigs for each service. In fact I can add one for the reboot service right now. It will be the most usable I think.

Reply:
true

Reply:
Done